File: src/main/java/no/nav/familie/integrasjoner/egenansatt/internal/EgenAnsattConsumerConfig.java
package no.nav.familie.integrasjoner.egenansatt.internal;


import no.nav.tjeneste.pip.egen.ansatt.v1.EgenAnsattV1;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties
public class EgenAnsattConsumerConfig {

    @Bean
    public EgenAnsattConsumer egenAnsattConsumer(EgenAnsattV1 egenAnsattV1) {
        return new EgenAnsattConsumer(egenAnsattV1);
    }
}

File: src/main/java/no/nav/familie/integrasjoner/client/soap/EgenAnsattConsumerConfig.kt
package no.nav.familie.integrasjoner.client.soap

import no.nav.tjeneste.pip.egen.ansatt.v1.EgenAnsattV1
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration @EnableConfigurationProperties
class EgenAnsattConsumerConfig {

    @Bean
    fun egenAnsattConsumer(egenAnsattV1: EgenAnsattV1): EgenAnsattConsumer {
        return EgenAnsattConsumer(egenAnsattV1)
    }
}